Gems/GradientSignal/Code/Source/Editor/EditorGradientBakerComponent.cpp

```diff
@@ -16,6 +21,176 @@
 
 namespace GradientSignal
 {
+    // Custom AZ::Job so that we can bake the output image asynchronously.
+    // We create the AZ::Job with isAutoDelete = false so that we can detect
+    // when the job has completed, which means we need to handle its deletion.
+    BakeImageJob::BakeImageJob(
+        const GradientBakerConfig& configuration,
+        const AZ::IO::Path& fullPath,
+        AZ::Aabb inputBounds,
+        AZ::EntityId boundsEntityId
+        )
+        : AZ::Job(false, nullptr)
+        , m_configuration(configuration)
+        , m_outputImageAbsolutePath(fullPath)
+        , m_inputBounds(inputBounds)
+        , m_boundsEntityId(boundsEntityId)
+    {
+    }
+
+    void BakeImageJob::Process()
+    {
+        // Get the actual resolution of our image.  Note that this might be non-square, depending on how the window is sized.
+        const int imageResolutionX = aznumeric_cast<int>(m_configuration.m_outputResolution.GetX());
+        const int imageResolutionY = aznumeric_cast<int>(m_configuration.m_outputResolution.GetY());
+
+        // The TGA and EXR formats aren't recognized with only single channel data,
+        // so need to use RGBA format for them
+        int channels = 1;
+        if (m_outputImageAbsolutePath.Extension() == ".tga" || m_outputImageAbsolutePath.Extension() == ".exr")
+        {
+            channels = 4;
+        }
+
+        int bytesPerPixel = 0;
+        OIIO::TypeDesc pixelFormat = OIIO::TypeDesc::UINT8;
+        switch (m_configuration.m_outputFormat)
+        {
+        case OutputFormat::R8:
+            bytesPerPixel = 1;
+            pixelFormat = OIIO::TypeDesc::UINT8;
+            break;
+        case OutputFormat::R16:
+            bytesPerPixel = 2;
+            pixelFormat = OIIO::TypeDesc::UINT16;
+            break;
+        case OutputFormat::R32:
+            bytesPerPixel = 4;
+            pixelFormat = OIIO::TypeDesc::FLOAT;
+            break;
+        }
```

Comment:
Nit: Might want to consider a default case that asserts.

Reply:
Good point will add